src/OpenApi/src/Services/OpenApiDocumentService.cs

```diff
@@ -31,7 +54,42 @@ public Task<OpenApiDocument> GetOpenApiDocumentAsync()
             Paths = GetOpenApiPaths(capturedTags),
             Tags = [.. capturedTags]
         };
-        return Task.FromResult(document);
+        await ApplyTransformersAsync(document, cancellationToken);
+        return document;
+    }
+
+    private async Task ApplyTransformersAsync(OpenApiDocument document, CancellationToken cancellationToken)
+    {
+        var documentTransformerContext = new OpenApiDocumentTransformerContext
+        {
+            DocumentName = documentName,
+            ApplicationServices = serviceProvider,
+            DescriptionGroups = apiDescriptionGroupCollectionProvider.ApiDescriptionGroups.Items,
+        };
+        // Use index-based for loop to avoid allocating an enumerator with a foreach.
+        for (var i = 0; i < _options.DocumentTransformers.Count; i++)
+        {
+            var transformer = _options.DocumentTransformers[i];
+            try
+            {
+                await transformer.TransformAsync(document, documentTransformerContext, cancellationToken);
+            }
+            finally
+            {
+                if (transformer is IDisposable disposable)
```

Comment:
Aren't these transformers effectively singletons because they come from the options object? So disposing them every use would be odd?